### ISSUE TITLE
Error with calculating log-probs for mask tokens

### ISSUE THREAD

**Issue by venkatasg** [babylm/evaluation-pipeline#12]
In `huggingface.py` the `_prepare_text` class method takes as input `token_ids`, and ought to return a list of `token_ids` with masks in each effective position. However, the code throws an error if the tokenizer that is used is setup to not add `bos` tokens. [Line 694](https://github.com/babylm/evaluation-pipeline/blob/4a914d988a85bf78098068a330b0a1b2ce80c41a/lm_eval/models/huggingface.py#L694) excludes the first and last token. However, the first token might need to be masked too if it is an effective token. Moreover, our tokenizer has an `eos_token` only, but line 687 doesn't check for that. I think that line should be changed to:

```
 effective_token_ids = [token for token in token_ids if token != self.tokenizer.pad_token_id and token != self.tokenizer.cls_token_id and token != self.tokenizer.sep_token_id and token != self.tokenizer.bos_token_id and token != self.tokenizer.eos_token_id]
```

to be applicable to all possible models.

**Comment by venkatasg** [comment]
I also need to comment line 694 and change line 691 to the following:

```
mask_indices = [[mask_pos] for mask_pos in range(effective_length)]
```

Originally, it had `effective_length+2`. No idea why you need to add 2 then immediately remove it?

**Comment by venkatasg** [comment]
The offset needs to be set to zero as well. So line 701 needs to change to:

```
masked_tensors.append((torch.stack(token_ids_masked_list), torch.stack(attention_masked_list), effective_token_ids, len(mask_indices), 0))
```

**Comment by venkatasg** [comment]
This issue still persists and the blimp eval code doesn't run for my encoder type model in the latest commit.
